docs(install): add mise alternative installation method (#1290)
hetznercloud/cli: docs/tutorials/setup-hcloud-cli.md

brew install hcloud
```

### 1.4 Installation using mise-en-place

> On Linux, macOS and Windows, if you are using [mise-en-place](https://github.com/jdx/mise), the polyglot tool version manager you can use a command like this to install `hcloud` cli:

Reviewer: ```suggestion
On Linux, macOS and Windows, if you are using [mise-en-place](https://github.com/jdx/mise), you can use this command to install `hcloud` globally:
```

```bash
mise use -g hcloud@latest
```

> *Note: This repository is using `mise` as explained in [CONTRIBUTING](../../CONTRIBUTING.md)*

Reviewer: If we want to add this, we should use the notation GitHub uses for notes:
```suggestion
> [!NOTE]  
> This repository is using `mise` as explained in [CONTRIBUTING](../../CONTRIBUTING.md)
```

Although I am not sure how relevant this is to the reader, since we only use mise in our internal build/development processes.
